### PR TITLE
Different client and configs for HBase 2.x

### PR DESCRIPTION
**Pull request by Arun-Krishna**

### CODE REVIEW

File: bigtable/bigtable.sh [hunk 1]
```diff
@@ -109,14 +108,82 @@ EOF
     --clobber
 }
 
+function configure_bigtable_client_2x() {
+  cat <<EOF >hbase-site.xml
```
Comment (medb) [inline bigtable/bigtable.sh:112]: We need to create it in temp directory;

```bash
local hbase_config
hbase_config=$(mktemp /tmp/hbase-site.xml-XXXX)
```

File: bigtable/bigtable.sh [hunk 2]
```diff
@@ -109,14 +108,82 @@ EOF
...
+HADOOP_CLASSPATH="${HADOOP_CLASSPATH}:/etc/hbase/conf"
+EOF
+
+  cat <<'EOF' >>/etc/spark/conf/spark-env.sh
+SPARK_DIST_CLASSPATH="${SPARK_DIST_CLASSPATH}:/usr/lib/hbase/*"
+SPARK_DIST_CLASSPATH="${SPARK_DIST_CLASSPATH}:/usr/lib/hbase/lib/*"
+SPARK_DIST_CLASSPATH="${SPARK_DIST_CLASSPATH}:/etc/hbase/conf"
+EOF
+
+  if [ "$hbase_version" == "2" ]
```
Comment (medb) [inline bigtable/bigtable.sh:157]: Use `[[` instead of `[`

File: bigtable/bigtable.sh [hunk 3]
```diff
@@ -109,14 +108,82 @@ EOF
...
+  then
+    configure_bigtable_client_2x || err 'Failed to configure big table 2.x client.'
+  else
+    configure_bigtable_client_1x || err 'Failed to configure big table 1.x client.'
+  fi
+}
+
+function get_hbase_major_version() {
```
Comment (medb) [inline bigtable/bigtable.sh:165]: We usually make conditions based on the Dataproc image version, it's available in `DATAPROC_VERSION` env variable.

File: bigtable/bigtable.sh [hunk 4]
```diff
@@ -109,14 +108,82 @@ EOF
...
+
 function main() {
   retry_apt_command "apt-get update" || err 'Unable to update packages lists.'
   retry_apt_command "apt-get install -y hbase" || err 'Unable to install HBase.'
 
-  install_bigtable_client || err 'Unable to install big table client.'
-  configure_bigtable_client || err 'Failed to configure big table client.'
+  local -r hbase_version=$(get_hbase_major_version)
+  if [ "$hbase_version" = "2" ]
+  then
```
Comment (medb) [inline bigtable/bigtable.sh:178]: Move then on the same line as `if` for consistency:

```bash
if [[ "$hbase_version" = "2" ]]; then
```

**Review by medb — CHANGES_REQUESTED** [review]
We will need to wait on new image release before merging this PR to make sure that GitHub presubmit passes.

File: bigtable/bigtable.sh (hunk 1 above)
Comment (Arun-Krishna) [reply bigtable/bigtable.sh]: Done.

File: bigtable/bigtable.sh (hunk 2 above)
Comment (Arun-Krishna) [reply bigtable/bigtable.sh]: Done.

File: bigtable/bigtable.sh (hunk 4 above)
Comment (Arun-Krishna) [reply bigtable/bigtable.sh]: Done.

**Review by animeshnandanwar — APPROVED** [review]
LGTM

**Pull request merged**